### ISSUE TITLE
Help to improve shallow model for citeseer more accure, link prediction

### ISSUE THREAD

**Issue by IRoyalX** [IRoyalX/Citeseer-shallow#1]
I have below code which is using citeseer dataset trained by a shallow model it accuracy is about 62% i need to improve it atleast up to 75% (goal 85%) keep it shallow
please help me if you can

`import networkx as nx
import numpy as np
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.metrics import accuracy_score
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC
from node2vec import Node2Vec
import os

# Select Dataset [ citeseer, cora, PubMed_div ] 
dataset = "citeseer"

os.system("cls")
print("Loading Dataset ...")
G = nx.read_edgelist(f"{dataset}.cites", create_using=nx.DiGraph())
node_data = np.loadtxt(f"{dataset}.content", dtype=str)
node_data = [data for data in node_data if data[0] in G.nodes()]
node_ids, node_features, node_labels = zip(*[(data[0], data[1:-1].astype(str), data[-1]) for data in node_data])
node_index = {node_id: i for i, node_id in enumerate(node_ids)}
nodes = [node_id for node_id, _ in sorted(node_index.items(), key=lambda item: item[1])]

os.system("cls")
# Generate walks using node2vec
node2vec = Node2Vec(G, dimensions=256, walk_length=70, num_walks=200, workers=4)
model = node2vec.fit(window=20, min_count=0, sg=1, epochs=20)
embeddings = np.array([model.wv[node] for node in nodes])

# Split the data into train and test sets (20% for test)
X_train, X_test, y_train, y_test = train_test_split(embeddings, node_labels, test_size=0.2, random_state=42)
Scale = StandardScaler()
X_train = Scale.fit_transform(X_train)
X_test = Scale.transform(X_test)

# best_params = {'C':8}
##### Grid Search
grid_search = GridSearchCV(SVC(kernel='rbf', gamma='scale', random_state=42), {'C': range(1, 11)}, cv=4)
grid_search.fit(X_train, y_train)
best_params = grid_search.best_params_
##### print("Best Hyperparameters:", best_params)

# Train SVM with best hyperparameters
eclf = SVC(kernel='rbf', C=best_params['C'], gamma='scale', random_state=42)
eclf = eclf.fit(X_train, y_train)

os.system("cls")
print("Accuracy:", accuracy_score(y_test, eclf.predict(X_test)))`
